### PR TITLE
Rewrite ChangeResolution filter to fix bugs and crashes

### PR DESCRIPTION
**Pull request by imikejackson**
Renames the filter to "Resample Image Geometry".

Also a code cleanup/fix that hits a bunch of misc filters was also applied in this because it was needed to get the unit test to pass.

Azure build bots are passing.

Related PRS:

https://github.com/BlueQuartzSoftware/SIMPLView/pull/71

https://github.com/BlueQuartzSoftware/SIMPL/pull/415

### CODE REVIEW

**Comment by imikejackson** [comment]
https://github.com/BlueQuartzSoftware/DREAM3DCI/pull/53
https://github.com/BlueQuartzSoftware/DREAM3DCI/pull/52

**Review by mmarineBlueQuartz — APPROVED** [review]

File: Source/Plugins/Sampling/SamplingFilters/ResampleImageGeom.cpp [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+          z = (i * m_Spacing[2]);
+          col = static_cast<size_t>(x / m_OrigSpacing[0]);
+          row = static_cast<size_t>(y / m_OrigSpacing[1]);
+          plane = static_cast<size_t>(z / m_OrigSpacing[2]);
+          index_old = (plane * m_OrigDims[1] * m_OrigDims[0]) + (row * m_OrigDims[0]) + col;
+          index = (i * m_CopyDims[0] * m_CopyDims[1]) + (j * m_CopyDims[0]) + k;
+          m_NewIndices[index] = index_old;
+        }
+      }
+    }
```
Comment (JDuffeyBQ) [inline Source/Plugins/Sampling/SamplingFilters/ResampleImageGeom.cpp]:
```suggestion
    for(size_t i = zStart; i < zEnd; i++)
    {
      for(size_t j = yStart; j < yEnd; j++)
      {
        if(m_Filter->getCancel())
        {
          return;
        }

        for(size_t k = xStart; k < xEnd; k++)
        {
          float x = (k * m_Spacing[0]);
          float y = (j * m_Spacing[1]);
          float z = (i * m_Spacing[2]);
          size_t col = static_cast<size_t>(x / m_OrigSpacing[0]);
          size_t row = static_cast<size_t>(y / m_OrigSpacing[1]);
          size_t plane = static_cast<size_t>(z / m_OrigSpacing[2]);
          size_t index_old = (plane * m_OrigDims[1] * m_OrigDims[0]) + (row * m_OrigDims[0]) + col;
          size_t index = (i * m_CopyDims[0] * m_CopyDims[1]) + (j * m_CopyDims[0]) + k;
          m_NewIndices[index] = index_old;
        }
      }
    }
```

File: Source/Plugins/Sampling/SamplingFilters/ResampleImageGeom.cpp [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+    size_t newIndicies_I = 0;
+    int32_t nComp = m_SourceData->getNumberOfComponents();
+
+    for(size_t i = start; i < end; i++)
+    {
+      newIndicies_I = m_NewIndices[i];
+      source = m_SourceData->getVoidPointer((nComp * newIndicies_I));
+      destination = m_DestinationData->getVoidPointer((m_DestinationData->getNumberOfComponents() * i));
+      ::memcpy(destination, source, m_SourceData->getTypeSize() * m_DestinationData->getNumberOfComponents());
+    }
```
Comment (JDuffeyBQ) [inline Source/Plugins/Sampling/SamplingFilters/ResampleImageGeom.cpp]:
```suggestion
    int32_t nComp = m_SourceData->getNumberOfComponents();

    for(size_t i = start; i < end; i++)
    {
      size_t newIndicies_I = m_NewIndices[i];
      const void* source = m_SourceData->getVoidPointer((nComp * newIndicies_I));
      void* destination = m_DestinationData->getVoidPointer((m_DestinationData->getNumberOfComponents() * i));
      std::memcpy(destination, source, m_SourceData->getTypeSize() * m_DestinationData->getNumberOfComponents());
    }
    }
```

File: Source/Plugins/Sampling/SamplingFilters/Utils/SamplingUtils.hpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ * @param cellFeatureAttributeMatrixPath
+ * @param featureIdsArrayPath
+ * @param saveAsNewDataContainer
+ */
+static void RenumberFeatures(AbstractFilter* filter, 
+              DataArrayPath newDataContainerPath, 
+              DataArrayPath cellAttributeMatrixPath, 
+              DataArrayPath cellFeatureAttributeMatrixPath,
+              DataArrayPath featureIdsArrayPath,
+              bool saveAsNewDataContainer)
```
Comment (JDuffeyBQ) [inline Source/Plugins/Sampling/SamplingFilters/Utils/SamplingUtils.hpp]:
```suggestion
void RenumberFeatures(AbstractFilter* filter, 
              const DataArrayPath& newDataContainerPath, 
              const DataArrayPath& cellAttributeMatrixPath, 
              const DataArrayPath& cellFeatureAttributeMatrixPath,
              const DataArrayPath& featureIdsArrayPath,
              bool saveAsNewDataContainer)
```

File: Source/Plugins/Sampling/SamplingFilters/ResampleImageGeom.cpp [hunk 4]
```diff
@@ -0,0 +1,744 @@
...
+  reader->closeFilterGroup();
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void ResampleImageGeom::initialize()
+{
+}
+
```
Comment (JDuffeyBQ) [inline Source/Plugins/Sampling/SamplingFilters/ResampleImageGeom.cpp]:
```suggestion
```

**Pull request merged**